openquake/hazardlib/gsim/nath_2012.py

```diff
@@ -0,0 +1,201 @@
+    """
+    Implements GMPE of Nath et. al (2012) for intraplate margin seismicity in
+    the Shillong Plateau of India at 25-45 km deph.
+
+    This model is based on stochastic simulation with a mean stress drop of 150
+    bars.
+
+    Verification of mean value data was done by digitizing Figure 11 using
+    http://arohatgi.info/WebPlotDigitizer/app/. Note that this independent
+    verification did not include magnitude dependence or standard deviations.
+
+    **Reference**
+
+    Page number citations in this documentation refer to:
+
+    Nath, S. K., Thingbaijam, K. K. S., Maiti, S. K., and Nayak, A. (2012).
+    Ground-motion predictions in Shillong region, northeast India. *Journal of
+    Seismology*, 16(3):475–488.
+    """
+
+    #: "studies on micro-earthquakes indicated that reverse faulting is
+    #: predominant in the region" (p. 476)
+    DEFINED_FOR_TECTONIC_REGION_TYPE = const.TRT.SUBDUCTION_INTERFACE
+
+    #: Set of :mod:`intensity measure types <openquake.hazardlib.imt>`
+    #: this GSIM can calculate. A set should contain classes from
+    #: module :mod:`openquake.hazardlib.imt`.
+    DEFINED_FOR_INTENSITY_MEASURE_TYPES = set([SA, PGA, PGV])
+
+    #: In simulations only the vertical component is estimated (see p. 479)
+    #: and the stochastic dataset is what the GMPE is based on, so this model
+    #: effectively predicts vertical motions.
+    DEFINED_FOR_INTENSITY_MEASURE_COMPONENT = const.IMC.VERTICAL
+
+    #: The only sigma is reported in the main coefficient table, Table 5 on
+    #: p. 483, and must be the total standard deviation.
+    DEFINED_FOR_STANDARD_DEVIATION_TYPES = set([const.StdDev.TOTAL])
+
+    #: Required site parameter Vs30 is used to determing the NEHRP
+    #: site class, and thus to choose site amplification coefficients
+    #: and site amplification stanard error from Table 5 on p. 208.
+    REQUIRES_SITES_PARAMETERS = set()
+
+    #: Sole required rupture parameter is magnitude, since faulting
+    #: style is not addressed.
+    REQUIRES_RUPTURE_PARAMETERS = set(('mag',))
+
+    #: It is noted that "r_rup is the fault-rupture distance in kilometers"
+    #: following equation (11) on p. 484.
+    REQUIRES_DISTANCES = set(('rrup',))
+
+    def get_mean_and_stddevs(self, sites, rup, dists, imt, stddev_types):
+        # pylint: disable=too-many-arguments
+        """
+        See :meth:`superclass method
+        <.base.GroundShakingIntensityModel.get_mean_and_stddevs>`
+        for specification of input and result values.
+
+        Implements equation (11) on p. 484:
+
+        ``ln(P) = c1 + c2*M + c3*(10 - M)^3 + c4*ln(R + c5*exp(c6*M)``
+        """
+
+        assert imt.__class__ in self.DEFINED_FOR_INTENSITY_MEASURE_TYPES
+        for stddev_type in stddev_types:
+            assert stddev_type in self.DEFINED_FOR_STANDARD_DEVIATION_TYPES
+
+        # obtain coefficients for required intensity measure type (IMT)
+        coeffs = self.COEFFS_BEDROCK[imt].copy()
```

Comment:
If you don't make a copy then the next line which "appends" the IMT-independent coefficients alters the original. See http://stackoverflow.com/questions/38987/how-can-i-merge-two-python-dictionaries-in-a-single-expression